qase-api/src/main/java/io/qase/api/utils/ManifestUtils.java

```diff
@@ -0,0 +1,62 @@
+package io.qase.api.utils;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ManifestUtils {
+
+    private static final String MANIFEST_RESOURCE_NAME = "META-INF/MANIFEST.MF";
+
+    public static Manifest getClassLoaderManifest(ClassLoader manifestLoader) throws IOException {
+        return new Manifest(openManifestStream(manifestLoader));
+    }
+
+    public static String tryGetManifestMainAttributeOrNull(
```

Comment:
Have you tried using the package class to get information about the implementation instead of reading it from the manifest?

Reply:
Yes, `Package`-based approach seems to be less verbose here [because](https://docs.oracle.com/javase/8/docs/api/java/lang/Package.html) the values mostly are stored in a manifest: 

> Typically, it is stored in the manifest that is distributed with the classes

I am going to utilize `Package`s here, if you don't object.